domain/keyupdater/state/controllerkeystate.go

```diff
@@ -49,10 +49,16 @@ WHERE key IN ($S[:])
 
 	keyValues := []keyValue{}
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		return tx.Query(ctx, stmt, sqlKeys).GetAll(&keyValues)
+		err := tx.Query(ctx, stmt, sqlKeys).GetAll(&keyValues)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				return nil
+			}
+		}
+		return errors.Trace(err)
```

Comment:
Honestly I really hate this style of error handling. Keep to a clear, easy to read pattern. This helps when someone goes to add more logic in the future.
```suggestion
		if errors.Is(err, sqlair.ErrNoRows) {
			return nil
		} else if err != nil {
			return err
		}
		return nil
```

`errors.Is` is a likely candidate for inlining, and there is almost no cost to the double not-nil check, and the compiler is likely to elide one of these anyway. If-else-if blocks are clear, linear, and less likely someone with come in and mess up with a merge/new added logic.